src/Lecture2.hs

```diff
@@ -158,16 +178,85 @@ You're free to define any helper functions.
 -}
 
 -- some help in the beginning ;)
+data Chest = Chest
+    { chestGold     :: Int
+    , chestTreasure :: Int
+    }
+    deriving (Show)
+
+add :: Chest -> Chest -> Chest
+add c1 c2 = Chest (chestGold c1 + chestGold c2)
+                  (chestTreasure c1 + chestTreasure c2)
+
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
-    , knightEndurance :: Int
+    { knightHealth     :: Int
+    , knightAttack     :: Int
+    , knightEndurance  :: Int
+    , knightExperience :: Int
+    , knightChest      :: Chest
```

Comment:
Not sure if it was in the scope of the exercise to extend the definition of `Knight`...